atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -156,6 +173,57 @@ public void tearDown() {
         puncherStore = null;
     }
 
+    private void waitToForceRefresh() throws InterruptedException {
+        Thread.sleep(2 * SMALL_REFRESH_MILLIS);
+    }
+
+    private int getShardCountFromRuntime() {
+        return runtimeSupplier.get().shards();
+    }
+
+    @Test
+    public void secondaryQueueReadsFirstQueueLastSweptTimestamp() {
+        // Second queue creation and initialization
+        MetricsManager secondQueueManager = MetricsManagers.createForTests();
+        TargetedSweeper secondQueue =
+                TargetedSweeper.createUninitializedForTest(secondQueueManager, runtimeSupplier::get);
+        secondQueue.initializeWithoutRunning(timestampsSupplier, timelockService, spiedKvs, txnService, mockFollower);
+
+        // First queue's enqueue/sweep
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS);
+        sweepNextBatchForAllShards(CONSERVATIVE);
+
+        // Second queue Last Swept TS metric update
```

Comment:
nit: not entirely convinced this comment adds much :)

This applies to most of your comments in this test - do they tell you anything that's not immediately obvious from the code? If they do, keep them. If, in this case you have

// update second queue last swept ts metric
secondQueue.updateLastSweptTsMetric

then you don't need them.

There are multiple schools of thought for commenting, but generally I like to have comments that actively describe why something non-obvious is happening (especially if it was non-obvious to you!) that describe the _why_, not the _how_